compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaIntSigned16TypeSymbol.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.ballerina.compiler.api.impl.symbols;
+
+import io.ballerina.compiler.api.ModuleID;
+import io.ballerina.compiler.api.symbols.IntSigned16TypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BIntSubType;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.compiler.util.Names;
+
+/**
+ * Represents the int:Signed16 type descriptor.
+ *
+ * @since 2.0.0
+ */
+public class BallerinaIntSigned16TypeSymbol extends AbstractTypeSymbol implements IntSigned16TypeSymbol {
+
+    public BallerinaIntSigned16TypeSymbol(CompilerContext context, ModuleID moduleID, BIntSubType signed16Type) {
+        super(context, TypeDescKind.INT, moduleID, signed16Type);
```

Comment:
Same as the Char type, will it be possible to have a subtype kind? Otherwise in order to distinguish we will have to have an instance of check

Reply:
We can include. The downside would be if we want to check if the type is an integer kind, we'll have to consider all the types and/or provide a util to check if the type is an integer kind

Reply:
My suggestion is, that we keep a separate API for the subtype. We have to consider this when we order the suggestions with the priority

Reply:
Added the discussed APIs to the respective type symbols.